jwst/spectral_leak/spectral_leak_step.py

```diff
@@ -0,0 +1,106 @@
+#! /usr/bin/env python
+
+from stdatamodels.jwst import datamodels
+from jwst.datamodels import ModelContainer
+from ..stpipe import Step
+import numpy as np
+
+from . import spectral_leak
+
+__all__ = ["SpectralLeakStep"]
+
+
+class SpectralLeakStep(Step):
+    """
+    The MIRI MRS has a spectral leak in which 6 micron light leaks into the
+    12 micron channel.  This step applies a correction to the 12 micron channel.
+    """
+
+    class_alias = "spectral_leak"
+
+    reference_file_types = ['mrsptcorr']
+
+    def process(self, input):
+        """Execute the step.
+
+        Parameters
+        ----------
+        input : container of models containing 1-D extracted spectra
+
+        Returns
+        -------
+        JWST DataModel
+            The corrected data model. This will be "input" if the step is skipped,
+            otherwise it will be a corrected 1D extracted spectrum that contains
+            the MRS channel 3B range.
+        """
+
```

Comment:
@drlaw1558  could you confirm that we will run this step on all MRS data and no skip is needed. Since there is no way to tell what channels or bands we have until we get into the step the step decides if the correct channel/bands exist and then applies the correction to the ch3 data

Reply:
Confirm that we want it ON by default for all POINT source data.  Since the POINT/EXTENDED decision fork is within the code itself, the code would always need to run.  Presumably this won't cause an issue for any other spec modes?